mempool/sorted_mempool.go

```diff
@@ -84,48 +84,48 @@ func (sm *SortedMempool) SetMinVal(val uint64) []Item {
 	return removed
 }
 
-// PopMin removes the minimum value in sm.
-func (sm *SortedMempool) PeekMin() Item {
+// PeekMin returns the minimum value in sm.
+func (sm *SortedMempool[T]) PeekMin() (T, bool) {
 	if sm.minHeap.Len() == 0 {
-		return nil
+		return *new(T), false
 	}
-	return sm.minHeap.items[0].tx
+	return sm.minHeap.items[0].item, true
 }
 
 // PopMin removes the minimum value in sm.
-func (sm *SortedMempool) PopMin() Item {
+func (sm *SortedMempool[T]) PopMin() (T, bool) {
 	if sm.minHeap.Len() == 0 {
-		return nil
+		return *new(T), false
```

Comment:
One of the annoying things about generics is you can't do a `nil` check against T

Reply:
Otherwise you need to do this (but you need to assume `T` is a pointer, which it may not be.
![image](https://user-images.githubusercontent.com/13023275/220239832-93388796-c58c-4af4-880e-e020219303bf.png)